src/main/java/spoon/refactoring/Issue.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.refactoring;
+
+public interface Issue {

```

Comment:
Do you mean to throw exception directly instead of collecting of issues and then throw an exception? 
I would like to support use case where list of issues is better then one exception on the first issue.
The idea is like this:
1) create an refactoring
2) call Refactoring#getIssues()
3) if there are some issues, then solve them automatically, by applying other refactorings
4) call Refactoring#getIssues() again
5) if there are still some issues then skip this refactoring
6) if there are no issues, then process refactoring.